test/src/sdk-harness/test_projects/result_option_expect/src/main.sw

```diff
@@ -0,0 +1,32 @@
+contract;
+
+abi MyContract {
+    fn option_test(should_revert: bool);
+    fn result_test(should_revert: bool);
+}
+
+enum Error {
+    Test: (),
+}
+
+impl MyContract for Contract {
+    fn option_test(should_revert: bool) {
```

Comment:
Could these be broken up into separate functions rather than a bool that controls which test you are running?